### PR TITLE
[ML] Add _meta field to data frame analytics config

### PR DESCRIPTION
**Pull request by droberts195**
This PR adds a new field, `_meta`, to the data frame analytics configuration.

The `_meta` field stores an arbitrary key-value map. Keys are strings. Values are arbitrary objects
(possibly also maps).

The `_meta` field can be updated using the data frame analytics `_update` endpoint.

### CODE REVIEW

File: docs/reference/ml/ml-shared.asciidoc [hunk 1]
```diff
@@ -1311,6 +1311,11 @@ The maximum number of decision trees in the forest. The maximum value is 2000.
 By default, this value is calculated during hyperparameter optimization.
 end::max-trees-trained-models[]
 
+tag::meta[]
+Advanced configuration option. Contains custom meta data about the job. For
```
Comment (przemekwitek) [inline docs/reference/ml/ml-shared.asciidoc:1315]: Not sure, should be "meta data" or "metadata"?

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsConfigUpdateTests.java [hunk 2]
```diff
@@ -59,6 +60,9 @@ public static DataFrameAnalyticsConfigUpdate randomUpdate(String id) {
         if (randomBoolean()) {
             builder.setMaxNumThreads(randomIntBetween(1, 20));
         }
+        if (randomBoolean()) {
```
Comment (przemekwitek) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsConfigUpdateTests.java:63]: Would it make sense to also add a separate unit test for merging the update? Analogous to `testMergeWithConfig_UpdatedDescription`?

Comment (droberts195) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsConfigUpdateTests.java:63]: Good point. The new test reveals that merge doesn't work correctly.

**Review by przemekwitek — APPROVED** [review]
LGTM

**Pull request merged**